File: solutions/r/anagram/1/anagram.R
anagram <- function(subject, candidates) {
  anagrams <- c()
  subject_sorted <- tolower(paste(sort(unlist(strsplit(subject, ""))), collapse = ""))
  for(candidate in candidates) {
    if(tolower(subject) != tolower(candidate)) {
      candidate_sorted <- tolower(paste(sort(unlist(strsplit(candidate, ""))), collapse = ""))
      if(subject_sorted==candidate_sorted) {
        anagrams <- append(anagrams, candidate)
      }
    }
  }
  return(anagrams)
}

File: solutions/julia/anagram/1/anagram.jl
function detect_anagrams(subject, candidates)
    subject = subject |> lowercase
    cleaned = lowercase.(candidates)
    return candidates .^ (anagram.(subject,cleaned)) |> filter(x->!isempty(x))
end

function anagram(first,second)
    first == second && return 0
    base = first |> collect |> sort |> join
    compare = second |> collect |> sort |> join
    return base == compare ? 1 : 0
end
